docs/generator.py

```diff
@@ -132,34 +135,51 @@ class ModelDocGenerator(ApiSpecDocGenerator):
 
     MODEL_TEMPLATE = 'model.md.j2'
 
-    def generate_doc_files(self, dest_dir, include_models=None):
-        model_dir = os.path.join(dest_dir, 'models')
+    def __init__(self, template_dir, template_ctx, api_spec):
+        super().__init__(template_dir, template_ctx, api_spec)
+        self._model_index = []
+        self._model_template = self._jinja_env.get_template(self.MODEL_TEMPLATE)
+        self._model_dir = None
+
+    def _process_single_model(self, model_name, operations):
+        model_api_spec = self._api_spec[SpecProp.MODELS].get(model_name, {})
+        displayed_model_name = self._get_display_model_name(model_name)
+        model_spec = ModelSpec(
+            name=displayed_model_name,
+            description=model_api_spec.get(PropName.DESCRIPTION, ''),
+            properties=model_api_spec.get(PropName.PROPERTIES, {}),
+            operations=[
+                {
+                    "name": op_name,
+                    "tag": op_spec[OperationField.TAGS][0]
+                } for op_name, op_spec in operations.items()
+            ]
+        )
+        model_content = self._model_template.render(model=model_spec, **self._template_ctx)
+        self._write_generated_file(self._model_dir, displayed_model_name + self.MD_SUFFIX, model_content)
+        self._model_index.append(displayed_model_name)
+
+    def model_should_be_ignored(self, model_name, include_models):
```

Comment:
Let's make `model_should_be_ignored` protected here and in the superclass as it should not be a part of the public API.